### DISCUSSION TITLE
Why LSTM is used in Set2Set?

### DISCUSSION THREAD

**Discussion by deliciouscat** [pyg-team/pytorch_geometric#3994] [General]
Hi,
I'm studying about some of graph to vec methods, and i'm curious that why LSTM cell is used in [Set2Set](https://github.com/pyg-team/pytorch_geometric/blob/62bf0f3a3cb3a86b8c5ba2f1c30da3d8874dff6b/torch_geometric/nn/glob/set2set.py#L57). 

![image](https://user-images.githubusercontent.com/68888590/152290664-28d3ad27-f19d-41c8-8aee-0aa868f0e629.png)

According to my understanding(like the scheme above), it is reasonable to use dence layer for making query vector, because **q_star** already reflected **h**. 

May I know the intention to design a (RNN based) structure like this?

**Comment by rusty1s** [comment]
I can't really speak for the intention of the authors, but the LSTM is clearly used in the [respective paper](https://arxiv.org/pdf/1511.06391.pdf), see Eq. (3). IMO, it may help to better preserve state across iterations.
